dependency.lic

```diff
@@ -49,10 +49,51 @@ class ArgParser
   end
 
   def display_args(data)
-    data.each do |def_set|
-      respond ''
-      respond "  ;#{Script.current.name} " + def_set.map { |x| format_item(x) }.join(' ')
-      def_set.each { |x| respond "   #{(x[:display] || x[:name]).ljust(12)} #{x[:description]} #{x[:options] ? '[' + x[:options].join(', ') + ']' : ''}"; }
+    # This section displays help output for a script. It recognizes when a script call has been given
+    # with the wrong number or incorrect arguments, or when a script has been called via
+    # ;script-name help. It outputs an example script call, including arguments, with optional
+    # arguments enclosed in brackets. If an optional script_summary argument has been specified within
+    # the script, it also outputs a summary of the script. Further, if yaml settings have been
+    # outlined in base-help.yaml, it will output help information for the yaml settings used.
+    # Example output when all help settings have been specified:
+    #
+    # >;circlecheck help
+    #   SCRIPT SUMMARY: Checks your circle and skills to next circle. Optionally checks skills for a target circle.
+
+    #   SCRIPT CALL FORMAT AND ARG DESCRIPTIONS (arguments in brackets are optional):
+    #    ;circlecheck [debug] [mode] [target] [script_summary]
+    #     debug        Runs the script with debug messages.
+    #     mode         Only display requirements you don't meet [brief, short, next]
+    #     target       See requirements for a specific circle
+
+    #   YAML SETTINGS USED:
+    #     circlecheck_prettyprint: Tells circlecheck to use the pretty print formatting for output.
+    if Script.current.name != "bootstrap"
+      data.each do |def_set|
+        def_set
+          .select { |x| x[:name].to_s == "script_summary" }
+          .each { |x| respond " SCRIPT SUMMARY: #{x[:description]} " }
+        respond ''
+        respond " SCRIPT CALL FORMAT AND ARG DESCRIPTIONS (arguments in brackets are optional):"
+        respond "  ;#{Script.current.name} " + def_set.map { |x| format_item(x) unless x[:name].to_s == "script_summary"}.join(' ')
+        def_set
+          .reject { |x| x[:name].to_s == "script_summary" }
+          .each { |x| respond "   #{(x[:display] || x[:name]).ljust(12)} #{x[:description]} #{x[:options] ? '[' + x[:options].join(', ') + ']' : ''}" }
+        respond ''
+      end
+
+      # Display help output for settings used in the script. Relies on base-help.yaml.
+      yaml_data = get_data('help').to_h
+      unless yaml_data.select { |field, info| info["referenced_by"].include?(Script.current.name)}.empty?
+        respond " YAML SETTINGS USED:"
+        yaml_data.select { |field, info| info["referenced_by"].include?(Script.current.name)}
+                  .each do | field, info |
```

Comment:
🧹 Purely a stylistic comment, I think the majority of block arguments in our codebase don't include the space between the `|` and the variable name.

If that's the code convention, then this line perhaps should become `.each do |field, info|` ?

Reply:
Good catch.